tamr_client/datasets/record.py

```diff
@@ -0,0 +1,135 @@
+"""
+See https://docs.tamr.com/reference/record
+"The recommended approach for interacting with records is to use the :func:`~tamr_client.record.upsert` and
+:func:`~tamr_client.record.delete` functions for all use cases they can handle. For more advanced use cases, the
+underlying _update function can be used directly."
+"""
+import json
+from typing import Dict, Iterable
+
+import tamr_client as tc
+from tamr_client.types import JsonDict
+
+
+class PrimaryKeyNotFound(Exception):
+    """Raised when referencing a primary key by name that does not exist."""
+
+    pass
+
+
+def _update(
+    session: tc.Session, dataset: tc.Dataset, updates: Iterable[Dict]
+) -> JsonDict:
+    """Send a batch of record creations/updates/deletions to this dataset.
+    You probably want to use :func:`~tamr_client.record.upsert`
+    or :func:`~tamr_client.record.delete` instead.
+
+    Args:
+        dataset: Dataset containing records to be updated
+        updates: Each update should be formatted as specified in the `Public Docs for Dataset updates <https://docs.tamr.com/reference#modify-a-datasets-records>`_.
+
+    Returns:
+        JSON response body from server
+
+    Raises:
+        requests.HTTPError: If an HTTP error is encountered
+    """
+    stringified_updates = (json.dumps(update) for update in updates)
```

Comment:
Maybe try to use `typing.cast` to deal with typechecking here?

```suggestion
    from typing import cast, IO
    ...
    stringified_updates = (json.dumps(update) for update in updates)
    io_updates = cast(IO, stringified_updates)
    ...
    data = io_updates
```

Reply:
This fixes the problem.